mini-lessons/elm/readme.md

```diff
@@ -0,0 +1,120 @@
+6. __Time-traveling debugger:__ What if you could pause time and replay all recent user inputs? What if you could make a code change and watch the results replay without a page refresh? [Try it out](http://elm-lang.org/blog/time-travel-made-easy) and see for yourself!
+
+## Getting Started with Elm (5 - 10/50)
+
+### Install
+
+#### as a module
+
+```
+$ npm install -g elm
+```
+
+#### … or with the [installer](http://elm-lang.org/install)
+
+If you have any problem installing with node. You can use the installer which also makes everything super easy.
+
+#### Check everything installed
+
+Run `elm` from the command line and you should see output that looks like:
+
+```
+Elm Platform 0.16.0 - a way to run all Elm tools
+
+Usage: elm <command> [<args>]
+
+Available commands include:
+...
+```
+
+Make sure you have version 0.16.0 — the update was about two weeks ago and some syntax we will be talking about today has changed.
+
+### Elm REPL (5 - 15/50)
+
+Pair up and look at [From JavaScript?](http://elm-lang.org/docs/from-javascript) in the Elm docs. Play around with some simple operations in the Repl (don't worry about running into errors, we'll get working on the basics in just a second but read the error messages and see if anything looks familiar). If you get into building larger projects with Elm (which you should because it's awesome), the compiler and its error messages are one of the languages great aspects. It can tell an incredible amount about how your program works, where there are problems and why.
+
+#### importing [modules](http://elm-lang.org/docs/syntax#modules)
+
+In Elm, code is organized into modules which we can think of just like node modules. They give us namespaced libraries. To import a module use the keyword import followed by the module name: `import <module-name>` (e.g. `import List`). The module is then available in that scope and its functions and definitions are accessable namespaced under the module name `<module>.<func/def>` .
+
+```
+> "String"
+"String" : String
+> String.length "cat"
+-- NAMING ERROR ---------------------------------------------- repl-temp-000.elm
+
+Cannot find variable `String.length`.
+
+3│   String.length "cat"
+     ^^^^^^^^^^^^^
+The qualifier `String` is not in scope.
+
+
+> import String
+> String.length "cat"
+3 : Int
+```
+
+We can use the `exposing` keyword to make module contents accessable without the module name prefix.
+
+```
+> String.length "cat"
+3 : Int
+> length "doesn't know"
+-- NAMING ERROR ---------------------------------------------- repl-temp-000.elm
+
+Cannot find variable `length`
+
+6│   length "doesn't know"
+     ^^^^^^
+Maybe you want one of the following?
+
+    List.length
+    String.length
+
+
+> import String exposing (length)
+> length "knows"
+5 : Int
+```
```

Comment:
https://github.com/ga-dc/curriculum/pull/583#discussion-diff-47724834

Update really only addresses the second point. Immutable data I can talk about and we'll do naming in core language. Maybe just not introduce the REPL until the Core language?